api/clients/codecs/default_blob_encoding_codec.go

```diff
@@ -0,0 +1,52 @@
+package codecs
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda/encoding/utils/codec"
+)
+
+type DefaultBlobEncodingCodec struct{}
+
+var _ BlobCodec = DefaultBlobEncodingCodec{}
+
+func (v DefaultBlobEncodingCodec) EncodeBlob(rawData []byte) ([]byte, error) {
+	// encode blob encoding version byte
+	codecBlobHeader := EncodeCodecBlobHeader(byte(DefaultBlobEncoding), uint32(len(rawData)))
+
+	// append raw data
+	encodedData := append(codecBlobHeader, rawData...)
+
+	// encode all data modulo bn254
+	encodedData = codec.ConvertByPaddingEmptyByte(encodedData)
+
+	return encodedData, nil
+}
+
+func (v DefaultBlobEncodingCodec) DecodeBlob(encodedData []byte) ([]byte, error) {
+	// decode raw data modulo bn254
+	decodedData := codec.RemoveEmptyByteFromPaddedBytes(encodedData)
+
+	versionByte, length, err := DecodeCodecBlobHeader(decodedData[:31])
+	if err != nil {
+		return nil, err
+	}
+
+	if DefaultBlobEncoding != BlobEncodingVersion(versionByte) {
```

Comment:
I'm wondering what this version byte tells us that we don't already know from the above steps.

Reply:
addressed here https://github.com/Layr-Labs/eigenda/pull/583#discussion_r1619430577